extensions/amp-analytics/0.1/test/test-transport.js

```diff
@@ -48,69 +57,193 @@ describe('amp-analytics.transport', () => {
+  });
+
   it('prefers beacon over xhrpost and image', () => {
-    setupStubs(true, true);
-    sendRequest(window, 'https://example.com/test', {
+    setupStubs(true, true, true, true);
+    transport.sendRequest(window, 'https://example.com/test', {
       beacon: true, xhrpost: true, image: true,
     });
-    assertCallCounts(1, 0, 0);
+    assertCallCounts(0, 1, 0, 0);
   });
 
   it('prefers xhrpost over image', () => {
-    setupStubs(true, true);
-    sendRequest(window, 'https://example.com/test', {
+    setupStubs(true, true, true, true);
+    transport.sendRequest(window, 'https://example.com/test', {
       beacon: false, xhrpost: true, image: true,
     });
-    assertCallCounts(0, 1, 0);
+    assertCallCounts(0, 0, 1, 0);
   });
 
   it('reluctantly uses image if nothing else is enabled', () => {
-    setupStubs(true, true);
-    sendRequest(window, 'https://example.com/test', {
+    setupStubs(true, true, true, true);
+    transport.sendRequest(window, 'https://example.com/test', {
       image: true,
     });
-    assertCallCounts(0, 0, 1);
+    assertCallCounts(0, 0, 0, 1);
   });
 
   it('falls back to xhrpost when enabled and beacon is not available', () => {
-    setupStubs(false, true);
-    sendRequest(window, 'https://example.com/test', {
+    setupStubs(false, false, false, true);
+    transport.sendRequest(window, 'https://example.com/test', {
       beacon: true, xhrpost: true, image: true,
     });
-    assertCallCounts(1, 1, 0);
+    assertCallCounts(0, 1, 1, 0);
   });
 
   it('falls back to image when beacon not found and xhr disabled', () => {
-    setupStubs(false, true);
-    sendRequest(window, 'https://example.com/test', {
+    setupStubs(false, false, false, true);
+    transport.sendRequest(window, 'https://example.com/test', {
       beacon: true, xhrpost: false, image: true,
     });
-    assertCallCounts(1, 0, 1);
+    assertCallCounts(0, 1, 0, 1);
   });
 
   it('falls back to image when beacon and xhr are not available', () => {
-    setupStubs(false, false);
-    sendRequest(window, 'https://example.com/test', {
+    setupStubs(false, false, false, false);
+    transport.sendRequest(window, 'https://example.com/test', {
       beacon: true, xhrpost: true, image: true,
     });
-    assertCallCounts(1, 1, 1);
+    assertCallCounts(0, 1, 1, 1);
+  });
+
+  it('must create xframe before sending message to it', () => {
+    expect(() => {
+      transport.sendRequest(window, 'https://example.com/test', {
+        iframe: 'https://example.com/test',
+      });
+    }).to.throw(/send message to non-existent/);
+  });
+
+  it('reuses cross-domain iframe', () => {
+    const config = {
+      iframe: 'https://example.com/test',
+    };
+    sandbox.spy(transport, 'useExistingOrCreateCrossDomainIframe_');
+    sandbox.spy(transport, 'createCrossDomainIframe_');
+    transport.processCrossDomainIframe(window, config);
+
+    assert(transport.useExistingOrCreateCrossDomainIframe_.calledOnce);
```

Comment:
Hmm. I thought they were equivalent, but just tested with assert(false) and expect(false).to.be.true and I see now that the latter has a more informative message.
Converted these to look like:
    expect(transport.createCrossDomainIframe.calledOnce).to.be.true;